Assignment-1

1. Write a Python expression that adds two numbers: 7 and 5.
a= 7
b= 5
2. Subtract 4 from 9 using Python.
a= 9
b =4
3. Multiply 6 by 8 in Python.
 a= 6
 b =8

In [1]:
print(7+5)
print(9-4)
print(6*8)

12
5
48


4. Perform floor division (integer division) on 15 by 2. 
5. Find the remainder when 22 is divided by 5.
6. Find the remainder when -22 is divided by 8.
7. Find the remainder when 2 is divided by 7.

In [2]:
print(15//2)
print(22%5)
print(-22%8)
print(2%7)


7
2
2
2


8. Addition Assignment,
If x = 10, add 5 to x using an assignment operator.
9. Subtraction Assignment,
If y = 20, subtract 3 from y using an assignment operator.
10. Multiplication Assignment,
If z = 7, multiply z by 2 using an assignment operator.
11. Division Assignment
 If w = 40, divide w by 4 using an assignment operator.

In [3]:
x=10
x+=5
print(x)

y=20
y-=3
print(y)
z=7
z*=2
print(z)
w=40
w/=4
print(w)


15
17
14
10.0


12. Determine if 8 is greater than 6.
13. Determine if 5 is less than 10. 
14. Check if 7 is greater than or equal to 7.
15. Calculate the result of (3 + 5) * 2 - 4 / 2 % 3 using python 
operators (Use pen copy to solve question)
16. Evaluate the expression (7 ** 2 + 6 // 3) * (8 % 3 - 2)


In [4]:
print(8>6)
print(5<10)
print(7>=7)
print((3 + 5) * 2 - 4 / 2 % 3)
print((7 ** 2 + 6 // 3) * (8 % 3 - 2))

True
True
True
14.0
0


17. a = 8
 b = a * 2
 c = b // 3 + a
 print(a, b, c)
18. p = 10
 q = p - 4
 r = q + p * 2
 print(p, q, r)
19. m = 3
 n = m ** 2
 o = n % m + 1
 print(m, n, o)
20. a = 12
 b = a / 3
 c = b ** 2
 d = (a + b + c) % 5
 print(a, b, c, d)
21. c, d = 12, 6
 c, d = d + 4, c - 2
 d, c = c // 3, d * 3
 print(c, d +1)
22. m, n a = 8, 3,-4
 m, n = n * 2, m+ a - 5
 n, m = m + 4, n // 2
 print(m, n)
23. a, b = 5, 8
 a += b
 b -= 2
 a, b = b, a - 4
 print(a, b)
24. x, y = 12, 3
 y = x // 2
 x *= 3
 x, y = y + 1, x - 5
 print(x, y)
25. c, d = 8, 20
 d = c * 2
 c += 5
 c, d = d - 7, c // 3
 print(c, d)

In [5]:
a = 8
b = a * 2
c = b // 3 + a
print(a, b, c)

p = 10
q = p - 4
r = q + p * 2
print(p, q, r)

m = 3
n = m ** 2
o = n % m + 1
print(m, n, o)

a = 12
b = a / 3
c = b ** 2
d = (a + b + c) % 5
print(a, b, c, d)

c, d = 12, 6
c, d = d + 4, c - 2
d, c = c // 3, d * 3
print(c, d +1)

m, n, a = 8, 3,-4
m, n = n * 2, m+ a - 5
n, m = m + 4, n // 2
print(m, n)

a, b = 5, 8
a += b
b -= 2
a, b = b, a - 4
print(a, b)

x, y = 12, 3
y = x // 2
x *= 3
x, y = y + 1, x - 5
print(x, y)

c, d = 8, 20
d = c * 2
c += 5
c, d = d - 7, c // 3
print(c, d)

8 16 13
10 6 26
3 9 1
12 4.0 16.0 2.0
30 4
-1 10
6 9
7 31
9 4


Input- output 

Q1. WAP to accept integer input from user. 

In [ ]:
x= int(input())

Q2. Write a python program to take input from user age then increase his/her age by 10 years. 
Sample Input : enter you age : 12 
Sample Output : 22 

In [18]:
x=float(input('enter your age'))
y=x+10
print(y)

22.0


Q3.  Take input user name ,  age  , height  then print output as “my name is (your name),my age is (you age), and height is (your age)” . 
sample Input: Enter you name : alice 
Enter your age : 10 
Enter your height : 3.5 
Sample Output: 
My name is alice, my age is 10 and height is 3.5 

In [17]:
x=input("enter your name")
y=int(input('enter your age'))
z=float(input('enter your hight'))
print('My name is',x,',','My age is',y,'and hight is',z)

My name is saquib , My age is 21 and hight is 5.9


Q4. How do you input a floating-point number from the user and store it in a variable called price? 

In [14]:
x=float(input('enter the no.'))
Price=x
print(id(x),id(Price))

2203065264368 2203065264368


Q5. Write a Python program that asks the user for two numbers and then prints their sum 

Sample Input : Enter 1st numbers : 3 
               Enter 2nd number :4 
Sampel Output : 7 

In [12]:
x=int(input('enter first no.'))
y=int(input('enter second no.'))
print(x+y)

7


Q6.Write a Python program that asks the user for their age and then prints a message stating their age. 

In [10]:
x=int(input("enter your age"))
print('age is', x)

age is 5


Q7.Write a Python program that asks the user for the radius of a circle and then prints the area of the circle. 

In [9]:
x=float(input("Enter radius"))
y=3.14*(x**2)
print(y)

78.5


Q8. How do you take a string input from the user and convert it to an integer? 

Sample input : ‘234’ 
Sample output : 234 

In [8]:
x=input("enter the value")
y=int(x)
print(y)

123


Q9. Write a Python program that asks the user for their height in feet and converts it to meters (1 foot = 0.3048 meters).

In [4]:
x=float(input("Enter hight in feet"))
y=x*0.3048
print(y,"meters")

1.524 meters


Q10. Write a Python program that takes three inputs from the user: two integers and one floating-point number. Your task is to print these three values in the following format: 
Integer1  Integer2  Float_Number 

Sample Input: 
Enter the first integer: 5 
Enter the second integer: 10 
Enter the floating-point number: 3.14 
Sample output: 
5 10 3.14 

In [3]:
x=int(input("Enter first integer"))
y=int(input("Enter second integer"))
z=float(input("Enter the floating-point number"))
print(x,y,z)

5 10 3.14


if else assignment 1



Q1. W.A.P  to check if a number is positive, negative. 

Input: enter a number : 8 
Output : positive  


  

In [3]:
x=int(input("Enter the number"))
if x<0 :
    print("Negative")
elif x>0 :
    print("Positive")
else:
    print("Zero")

Zero


Q2 W.A.P that takes 2 numbers input from user check their sum is greater then 10 or not . if greater than 10 print ‘greater’ otherwise print ‘less’ 

Input 
Enter 1 number: 3 
Enter 2 number : 5 
Output: 
less  



In [7]:
x=int(input("Enter first no."))
y=int(input("Enter second no"))
z=x+y
if z>10 :
    print("greater")
else:
    print("less")

less


Q3. WAP that takes 2 numbers from user and print largest number 

Input 
Enter a number: 78 
Enter a number:  98 
Output: 
98 



In [ ]:
x=int(input("Enter first no."))
y=int(input("Enter second no"))
if x>y :
    print(x)
else:
    print(y)

Q4. W.A.P  that takes a number check  it is divisible by 5 or not 

Input  
Enter a number : 90 
Output: 
Divisible  




In [10]:
x=int(input("Enter the no."))
if x%5==0 :
    print("divisible by 5")
else:
    print("not divisible by 5")

not divisible by 5


Q5. w.a.p that 4 angles checks whether it can form quadrilateral or not. 
Sum of all four angle is 360  



In [13]:
a=int(input("Enter angleA"))
b=int(input("Enter angleB"))
c=int(input("Enter angleC"))
d=int(input("Enter angleD"))
z=a+b+c+d
if z==360 :
    print("valid input")
else:
    print("not valid input")

valid input


Q6.  Determine the season based on month taken from user 

3 <= month <= 5; print "Spring" 
6 <= month <= 8 then print "Summer" 
9 <= month <= 11, then print "Autumn" 
12<= moth <= 2 then print "Winter" 

In [ ]:
m=int(input("Enter month in no."))
if 3<=m and m<=5 :
    print("Spring")
elif 6<=m and m<=8 :
    print("Summer") 
elif 9<=m and m<=11 :
    print("Autumn") 
else:
    print("Winter")

Winter


If else Assignment - 2
Q1. W.A.P to check if a number is positive, negative, or zero.
Input: enter a number: 8
Output: positive

In [19]:
x=int(input("Enter the number"))
if x>0 :
    print("positive")
elif x<0:
    print("Negative")
else :
    print("Zero")

Zero


Q2. W.A.P to check if a number is even or odd.
Input: 9
Output: odd

In [21]:
x=int(input("Enter the number"))
if x%2 == 0:
    print("even no")
else:
    print("odd no.")

even no


Q3. W.A.P to Determine grade based on marks.
marks >= 90: Grade A
 marks >= 80: Grade: B
 Marks >= 70: Grade C
 marks >= 60: Grade: D
 otherwise,
 Grade: E
Test Case 1:
Input: 50
Output: E
21 September 2024

In [26]:
x=int(input("Enter percentage"))
if 90<=x<=100 :
    print("GradeA")
elif x>=80 :
    print("GradeB")
elif x>=70 :
    print("GradeC")
elif x>=60 :
    print("GradeD")
else:
    print("GradeE")

GradeD


Q4. W.A.P to Check if a number is divisible by both 3 and 5.
Input: 15 
Output: Divisible

In [35]:
x=int(input("Enter the no."))
y=x%3
z=x%5
l=y==0 and z==0
if l  :
    print("divisible by both")
elif y==0 :
    print("Divisible by 3")
elif z==0 :
    print("Divisible by 3")
else:
    print("not divisible by 3 or 5")


Divisible by 3


Q5. program to determine if a person is a child, teenager, adult, or 
senior based on their age
age <= 12:"Child"
age <= 19: "Teenager" 
age <= 64: "Adult" 
age > = 64:"Senior"
Test Case 1:
Enter age: 8
Child
Test Case 2:
Enter age: 16
Teenager

In [40]:
age= int(input("Enter your age"))
if age>= 64 :
    print("senior")
elif age>=20 :
    print("Adult")
elif age>=12 :
    print("teenager")
else:
    print("child")

senior


Q6. Check if a year is a leap year.
Test case 1:
Input: 1600
Output: Leap year
Test Case 2:
Input: 1900
Output: Not a leap year.

In [53]:
x=int(input("Enter year"))
if x%4==0 :
    print("leap year")
else:
    print("not leap year")

leap year


Q7. W.A.P to find the largest of three numbers. 
Input: enter 3 number: 78, 56, 100
Output: greatest out of 3: 100


In [28]:
x=int(input("Enter the first no."))
y=int(input("Enter the second no."))
z=int(input("Enter the third no."))
if x>y and x>z :
    print(x)
elif y>x and y>z :
    print(y)
elif z>x and z>y :
    print(z)

345678


Q8. W.A.P Determine if a triangle is equilateral, isosceles, or scalene.
Input: enter 3 sides of a triangle 7, 7, 1 
Output: isosceles


In [57]:
a=int(input("Enter sideA"))
b=int(input("Enter sideB"))
c=int(input("Enter sideC"))
if a==b==c :
    print("equilateral triangle")
elif a==b!=c or a==c!=b or c==b!=a :
    print("isosceles triangle")
else:
    print("scalene triange")

isosceles triangle


Q9. Determine if a year is a century year.
Example of century year 1800, 2000, 2100 many more…

In [58]:
x=int(input("Enter year"))
if x%100==0 :
    print("Century year")
else:
    print("not century")

Century year


Q10.Write a Python program that accepts three numbers and check 
All numbers are equal or not.


In [ ]:
x=int(input("Enter 1st number"))
y=int(input("Enter 2nd number"))
z=int(input("Enter 3rd number"))
if x==y==z :
    print("all are equal")
else:
    print("No. are not equal")

Q10. Write a python program which takes x and y as a input and 
determine the quadrant of a point in a coordinate system as per the 
given information below:
 x> 0, y> 0: first quadrant
 x < 0, y > 0: second quadrant
 x< 0, y<0: third quadrant
 x> 0, y < 0: fourth quadrant
Test Case 1:
Input: x = 8 and y = -1
Output: fourth quadrant

In [59]:
x=int(input("Enter value of x"))
y=int(input("Enter value of y"))
if x>0 and y>0 :
    print("First quadrant")
elif x<0 and y>0:
    print("second quadrant")
elif x<0 and y<0 :
    print("third quadrant")
elif x>0 and y<0 :
    print("fourth quadrant")

fourth quadrant


Q12. Write a python program which takes three input and check if 
third number is within a given range between first and second 
number.
INPUT: LOWER RANGE: 10
 UPPER RANGE: 89
 INPUT_NUMBER: 77
OUTPUT: ‘YES’

In [57]:
a=int(input("Enter 1st value"))
b=int(input("Enter 2nd value"))
c=int(input("Enter 3rd value"))
if b<a<c :
    print("upper case =",c,"lowr case =",b)
elif c<a<b:
    print("upper case =",b,"lowr case =",c)
elif a<b<c :
    print("upper case =",c,"lowr case =",a)
elif c<b<a :
    print("upper case =",a,"lowr case =",c)
elif b<c<a :
    print("upper case =",a,"lowr case =",b)
elif a<c<b :
    print("upper case =",b,"lowr case =",a)

upper case = 70 lowr case = 55


Q13. Input 3 angle and determine if they form a triangle or not.

In [ ]:
a=int(input("Enter 1st side"))
b=int(input("Enter 2nd side"))
c=int(input("Enter 3rd side"))
if a+b>c and b+c>a and c+a>b :
    print("valid")
else :
    print("not valid")
    

Q14. program that reads 2 numbers and an arithmetic operator like 
+, -, *, /, % and display the computed result:
Example 
Enter the 1 number: 5
Enter the 2 number: 2
Enter the operator: *
Output: 10.0

In [ ]:
x=float(input("Enter first value"))
z=(input("enter operator = +,-,*,/,%"))
y=float(input("Enter second value"))
if z== "+" :
    print(x+y)
elif z=='-' :
    print(x-y)
elif z=='*' :
    print(x*y)
elif z=='/' :
    print(x/y)
elif z=='%' :
    print(x%y)
else:
    print("not valid operator")

-3.0


Q15. Write a python program to input cost price and selling price of a 
product and check profit or loss. Also calculate total profit or loss 
using if else.
Test case 1:
input cost price: 1000
Input selling price: 1500
Output:
Profit is 500

In [63]:
cp=float(input("Enter cost price"))
sp=float(input("Selling price"))
z=sp -cp
if sp>cp:
    print("profit")
    print("profit amout :",z)
else:
    print("loss")
    print("loss amount :", -z)

loss
loss amount : 40.0


Q16. Write a python program to input week number and print week 
day.
Test case 1:
Input: 1
Output: Monday
Test case 2:
Input: 7
Output: Sunday

In [65]:
w=int(input("Enter week no. from 1 to 7"))
if w==1:
    print("Monday")
elif w==2:
    print("Tuesday")
elif w==3:
    print("Wednesday")
elif w==4:
    print("Thrusday")
elif w==5:
    print("Friday")
elif w==6:
    print("Saturday")
elif w==7:
    print("Sunday")

Sunday


Q17. Write a python program that accepts three numbers from the 
user and check if numbers are in "increasing" or "decreasing" order.
Test Case 1:
Input: 10,17,20
Output: Increasing Order.
Test case 2:
Input: 90,69,36
Output: Decreasing Order

In [66]:
x=int(input("Enter 1st number"))
y=int(input("Enter 2nd number"))
z=int(input("Enter 3rd number"))
if x>y>z :
    print("Decreasing order")
elif x<y<z :
    print("Increasing order")
else:
    print("not in order ")

Increasing order


Q18. While purchasing certain items, a discount of 10% is offered if 
the quantity purchased is more than 100. If quantity and price per 
item are input through the keyboard, write a program to calculate 
the total expenses.

In [73]:
x=int(input("product 1st price"))
y=int(input("product 2nd price"))
z=int(input("product 3rd price"))
t= x+y+z

tp=t-(t*(10/100))


if t>=100:
    print("10% discount applied")
    print("payable price :", tp)
else:
    print("no discount applied")

no discount applied


Q19. Write a program to input electricity unit charges and calculate 
total electricity bill 
according to the given condition:
• For first 50 units Rs. 0.50/unit
• For next 150 units Rs. 0.75/unit
• For next 250 units Rs. 1.20/unit
• For unit above 450 Rs. 1.50/unit
• An additional surcharge of 20% is added to the bill
Test Case 1:
Input: 100 units
Expected Output:
Rs. 62.5
With Surcharge: Rs. 75
Test Case 3:
Input: 300 units
Expected Output:
Rs. 257.5
With Surcharge: Rs. 309

In [9]:
u=float(input("Enter unit of electricity "))
if u<=50 :
    a=u*0.50
    sa=a+(a*(20/100))
    print("Electricity bill :", a)
    print("with surcharge :", sa)
elif 50<u and u<=200 :
    b=(50*0.50)+((u-50)*0.75)
    sb=b+(b*(20/100))
    print("Electricity bill :", b)
    print("with surcharge :", sb)
elif 200<u and u<=450 :
    c=(50*0.50)+(150*0.75)+((u-200)*1.20)
    sc=c+(c*(20/100))
    print("Electricity bill :", c)
    print("with surcharge :", sc)
else:
    d=(50*0.50)+(150*0.75)+(250*1.50)+((u-450)*1.50)
    sd=d+(d*(20/100))
    print("Electricity bill :", d)
    print("with surcharge :", sd)

Electricity bill : 257.5
with surcharge : 309.0


In [ ]:
u=int(input("Enter the no of units "))
if u<=50:
    c=u*0.50
    print("Electricity bill :", c)
elif u<=200:
    c=25+(u-50)*0.75
elif u<=450:
    c=25+150*0.75+(u-200)*1.20
else:
    c=25+150*0.75+250*1.20+(u-450)*1.50

final_amt=c+c*0.2
print(final_amt)

309.0


Q20. Write a python program which accept the kilometres covered 
and calculate the bill according to the following criteria:
First 10 Km -> Rs11/km
Next 90Km -> Rs 10/km
After that -> Rs9/km
Test Case 1: 
Input: 50
Output: 510
Test Case 2:
Input: 100
Output: 1010

In [8]:
x=float(input('Enter kilometer'))
if  x<=10:
    a=x*11
    print('bill =',a)
elif x>10 and x<=100 :
    b=a+((x-10)*10)
    print('bill = ', b)
else:
    c=a+b+((x-100)*9)
    print('bill =',c)
    

bill =  510.0


for loop assignment-1

1. Write a python program that take a number from user then add all 
the natural number which are divisible by 3 or 5 both till that input 
number.
Sample input:
Enter a number: 10
Output: 33

In [3]:
y=0
x=int(input('Enter the no.'))
for i in range(1,x+1):
    if i%3==0 or i%5==0:
        y+=i
print(y)

33


2. Write a python program that takes input from user then print the 
table of the entered number using for loop
Example: If user enters ‘5’ then, 
Output: 5 x 1 = 5
 5 x 2=10
 5 x 3 =15
 5 x 4 =20
 5 x 5 = 25
 5 x 6 = 30
 5 x 7 = 35 
 5 x 8 = 40
 5 x 9 = 45
 5 x 10 =50


In [9]:
x=int(input('Enter the no.'))
for i in range(1,11):
    print(x,'*',i,'=',x*i)

5 * 1 = 5
5 * 2 = 10
5 * 3 = 15
5 * 4 = 20
5 * 5 = 25
5 * 6 = 30
5 * 7 = 35
5 * 8 = 40
5 * 9 = 45
5 * 10 = 50


3. Write a python program that print the given series.
Expected Output: 20, 16, 12, 8, 4, 0

In [13]:
for i in range(20,-1,-4):
    print(i,end=" ")

20 16 12 8 4 0 

4. Write a python program to generate all the factors of the given 
number. Using for loop
Input: 12
Output: 1, 2, 3, 4, 12

In [17]:
x=int(input('enter the no.'))
for i in range(1,x+1):
    if x%i==0:
        print(i,end=',')

1,2,3,4,6,12,

6. write a python program that takes number from user then prints 
count of even and odd number from 1 to till that number
Sample input:
Enter a number: 10
Output:
Count_of_even_no.s = 5
Count_of_odd_no.s = 5

In [1]:
evn=0
odd=0
x=int(input('Enter the no.'))
for i in range(1,x+1):
    if i%2==0:
        evn+=1

    else:
        odd+=1 
print(' Count_of_even_no.s',evn,'\n','Count_of_even_no.s',odd)
        


 Count_of_even_no.s 5 
 Count_of_even_no.s 5


7. Write a Python program to count how many numbers between 1 
and 200 are divisible by 5 but not by 2.

In [11]:
x=0
for i in range(1,201):
    if i%5==0 and i%2!=0:
        x+=1
print(x)

20


8. Write a Python program to find the reverse of a given number
using for loop. (between 100 and 999).
Input: 897
Output: 798

In [1]:
y=0
x=int(input('enter your no.'))
for i in range(3):
    r=x%10
    y=y*10+r
    x//=10
print(y)

321


5. Write a Python program to find the sum of digit of a given number
using for loop. (between 100 and 999).
Input: 123
Output: 6

In [22]:
z=0
x=int(input('enter the no.'))
for i in range(3):
    r=x%10
    
    z+=r
    x//=10
print(z)

18


9. Write a python program to input a number and calculate the 
‘double’ factorial using for loop.
Explanation: Double factorial,
• For an even integer n, the double factorial is the product of all 
even integers less than or equal to n.
• For an odd integer p, the double factorial is the product of all 
odd integers less than or equal to p.

In [18]:
x=1
n=int (input('enter your no.'))
for i in range(1,n+1):
    if n%2==0 and i%2==0:
        x*=i
        
    elif n%2!=0 and i%2!=0:
        x*=i
print(x)


105


In [26]:
x=int(input('enter your no.'))
ans=1
for i in range(x,0,-2):
    ans*=i
print(ans)

15


basic question 

In [3]:
for i in range(5,16):
    print(i,end=',')

5,6,7,8,9,10,11,12,13,14,15,

In [6]:
for i in range(2,25):
    if i%2==0 :
        print(i,end=',')

2,4,6,8,10,12,14,16,18,20,22,24,

In [7]:
for i in range(1,14):
    if i%2!=0:
        print(i,end=',')

1,3,5,7,9,11,13,

In [8]:
for i in range(1,21):
    print(i,end=',')

1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,

In [11]:
for i in range(20,0,-1):
    print(i,end=',')

20,19,18,17,16,15,14,13,12,11,10,9,8,7,6,5,4,3,2,1,

In [15]:
x=1
for i in range(2,13):
    x=i**2
    print(x,end=',')

4,9,16,25,36,49,64,81,100,121,144,

In [16]:
x=1
for i in range(1,7):
    x=i**3
    print(x,end=',')

1,8,27,64,125,216,

In [20]:
a=1
b=2
c=3
d=4
e=5
for i in range(1,51):
    a=i*a
    b=i*b
    c=i*c
    d=i*d
    e=i*e
    print(a,b,c,d,e)

1 2 3 4 5
2 4 6 8 10
6 12 18 24 30
24 48 72 96 120
120 240 360 480 600
720 1440 2160 2880 3600
5040 10080 15120 20160 25200
40320 80640 120960 161280 201600
362880 725760 1088640 1451520 1814400
3628800 7257600 10886400 14515200 18144000
39916800 79833600 119750400 159667200 199584000
479001600 958003200 1437004800 1916006400 2395008000
6227020800 12454041600 18681062400 24908083200 31135104000
87178291200 174356582400 261534873600 348713164800 435891456000
1307674368000 2615348736000 3923023104000 5230697472000 6538371840000
20922789888000 41845579776000 62768369664000 83691159552000 104613949440000
355687428096000 711374856192000 1067062284288000 1422749712384000 1778437140480000
6402373705728000 12804747411456000 19207121117184000 25609494822912000 32011868528640000
121645100408832000 243290200817664000 364935301226496000 486580401635328000 608225502044160000
2432902008176640000 4865804016353280000 7298706024529920000 9731608032706560000 12164510040883200000
51090942171709440000 102

In [24]:
for i in range(1,31):
    if i%3==0 or i%4==0:
        print(i,end=',')

3,4,6,8,9,12,15,16,18,20,21,24,27,28,30,

9. Write a for loop to print numbers from 1 to 40, but skip numbers that are divisible
by 7.


In [8]:
n=int(input())
if n%n>1==0 and n%n<n==0:
    print('not prime')
else:
    print('prime')

prime


In [5]:
4%3

1